ibm/mas_devops/roles/cos/tasks/providers/ibm/provision.yml

```diff
@@ -114,17 +116,19 @@
   when: cos_key_info.resource.credentials is defined
   set_fact:
     cos_password: "{{ cos_key_info.resource.credentials.apikey }};{{ cos_key_info.resource.credentials.resource_instance_id }}"
-
+    cos_hmac_access_key: "{{ cos_key_info.resource.credentials.cos_hmac_keys.access_key_id }}"
+    cos_hmac_secret_key: "{{ cos_key_info.resource.credentials.cos_hmac_keys.secret_access_key }}"
```

Comment:
cos_hmac_keys would be not set in the case where use_hmac is false. We should move these into a seperate task that has the condition `when: cos_key_info.resource.credentials is defined and use_hmac`

Reply:
Updated as per suggestion